helper-cli/src/main/kotlin/utils/PathExcludeGenerator.kt

```diff
@@ -62,6 +64,19 @@ internal object PathExcludeGenerator {
     }
 }
 
+private val REGEX_SPECIAL_CHARACTERS = listOf('.', '+', '*', '?', '^', '$', '(', ')', '[', ']', '{', '}', '|', '\\')
+private val REGEX_IGNORE_CASE_OPTION = setOfNotNull(RegexOption.IGNORE_CASE)
+private fun String.matchWildcard(value: String, ignoreCase: Boolean = false): Boolean {
```

Comment:
Can't you just treat the string as a path and then use the `FileMatcher` to match the string against a wildcard / glob pattern?

Reply:
I've updated the commit message in response to your above comment, which explains why I decided against that.

Reply:
Did you forget to push? I'm still seeing the old commit message (also incl. the "cleaar" typo).

Reply:
Pushed again, should be up-to-date now.

Reply:
Sorry, I'm having trouble understanding what you're trying to say in the commit message:

> Using wildcards has the advantage of matching more relevant cases

More relevant uses cases than *what*? What do you compare to?

> but it also introduces some false positives.

Why? How?

> as it is not yet clear whether these patterns should also end-up in the `PathExclude.pattern`

Why is that not clear yet? Who decides that?

The `pattern` property is documented as "A glob to match the path of the project definition file", so I always assumed it's perfectly valid to use full-blown globs here, not just wildcards.

I'm missing the overall picture of what you're trying to do. Are you trying to limit yourself *by convention* to just using wildcards instead of globs? And if so, why? Are you afraid that users might not be able to deal with the additional syntax that globs support over wildcards?

Reply:
> More relevant uses cases than _what_? What do you compare to?

compared to without wild cards.

> Why? How?

The matcher becomes broader so matches more dir names, maybe unintendedly some which should not be excluded.

> Why is that not clear yet? Who decides that?


Currently, the generated path excludes don't contain any `*` besides the `/**` suffix.
I plan to experiment with using the `*` also in generated path excludes. 
From these experiments I expect to gain some learnings, which will make it clearer to me.

Reply:
I went for glob expressions as they already give good results for the test data and are simpler. 
This may resolve the discussion, please take a look at the updated changes.